### PR TITLE
GAWB-1139: Enable Job Avoidance

### PR DESCRIPTION
**Pull request by ansingh7115**
What's inside:

* Addition of "Enable Job Avoidance" checkbox in Launch Analysis modal
* Statuses in the workflow details page for whether call caching was enabled/ disabled and (if enabled) whether each call was cache hit/miss 

- [ ] **Submitter**: Include the JIRA issue number in the PR description
- [ ] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * Add notes on what you've tested
- [ ] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [ ] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [ ] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [ ] Tell the tech lead (TL) that the PR exists if they wants to look at it
- [ ] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Product Owner** sign off
- [ ] **Assign to submitter** to finalize
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: src/cljs/main/broadfcui/page/workspace/method_configs/launch_analysis.cljs [hunk 1]
```diff
@@ -78,10 +78,18 @@
...
                                :onChange #(let [text (-> % .-target .-value clojure.string/trim)]
                                             (swap! state assoc :expression text))}))
+   [:div {:style {:marginTop "1em"}}
+    [comps/Checkbox
+    {:ref "callCache-check"
+     :label "Use Call Caching"
+     :initial-checked? true
+     :disabled-text (case (:protected-option @state)
+                      :not-loaded "Call Caching status has not finished loading."
+                      :not-available "This option is not available for your account."
```
Comment (helgridly) [inline src/cljs/main/broadfcui/page/workspace/method_configs/launch_analysis.cljs:89]: Nice!

File: src/cljs/main/broadfcui/page/workspace/monitor/common.cljs [hunk 2]
```diff
@@ -74,3 +74,13 @@
...
     :else (do (utils/log "Unknown call status: " status)
             failure-icon)))
+
+
+(defn call-cache-result [cacheStatus]
+   (if (= cacheStatus "ReadAndWriteCache") "Enabled" "Disabled")
+  )
+(defn format-call-cache [cacheHitMiss]
+   ;(get cacheHitMiss "Cache Hit" "Cache Miss")
+   (if (utils/contains cacheHitMiss "Cache Hit") "Hit" "Miss")
```
Comment (helgridly) [inline src/cljs/main/broadfcui/page/workspace/monitor/common.cljs]: I feel like this language is a little too technical. @vdauwera how do we talk to users about call caching?

File: src/cljs/main/broadfcui/page/workspace/monitor/common.cljs [hunk 3]
```diff
@@ -83,3 +83,13 @@
...
+  )
```
Comment (MattPutnam) [inline src/cljs/main/broadfcui/page/workspace/monitor/common.cljs]: Clean up closing parens and naming convention (`cacheStatus` -> `cache-status`, `cacheHitMiss` -> `cache-hit-miss`).

File: src/cljs/main/broadfcui/page/workspace/monitor/workflow_details.cljs [hunk 4]
```diff
@@ -122,6 +122,8 @@
               (create-field "ID" (data "jobId"))
               (let [status (data "executionStatus")]
                 (create-field "Status" (moncommon/icon-for-call-status status) status))
+              (when (= (data "Effective call caching mode") "ReadAndWriteCache")
+                (create-field "Cache Result" (moncommon/format-call-cache (utils/cljslog (data "Call caching read result")))))
```
Comment (MattPutnam) [inline src/cljs/main/broadfcui/page/workspace/monitor/workflow_details.cljs]: remove log before merge

**Review by helgridly — APPROVED** [review]
make sure Alex looks at the UI

File: src/cljs/main/broadfcui/page/workspace/monitor/workflow_details.cljs [hunk 5]
```diff
@@ -149,7 +151,9 @@
                                                    bucketName "/" submission-id "/"
                                                    workflow-name "/" (workflow "id") "/")})))
     (let [status (workflow "status")]
-      (create-field "Status" (moncommon/icon-for-wf-status status) status))
+      (create-field "Status" (moncommon/icon-for-wf-status status)))
+    (let [call-cache-status (get (first (first (vals (workflow "calls")))) "Effective call caching mode")]
```
Comment (MattPutnam) [inline src/cljs/main/broadfcui/page/workspace/monitor/workflow_details.cljs]: This is an awful lot of digging into a data structure.  Is that how it has to be?

If so, I'd recommend using the thread-first macro to make this easier to read:
```clojure
(-> (workflow "calls") vals first first (get "Effective call caching mode"))
;; or
(-> (vals (workflow "calls")) first first (get "Effective call caching mode"))
```
Depending on where you think the logical starting place of the chain is.

**Pull request merged**